exams/views_test.py

```diff
@@ -13,36 +22,109 @@ def test_success(self):
 
     def test_logout(self):
         """
         Test /pearson/logout URL
         """
         response = self.client.get('/pearson/logout/')
-        assert response.status_code == 302
+        assert response.status_code == status.HTTP_302_FOUND
         assert response.url == "/dashboard?exam=logout"
 
     def test_not_found(self):
         """
         Test a URL under /pearson that doesn't exist
         """
         response = self.client.get('/pearson/other/')
-        assert response.status_code == 404
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+@ddt.ddt
+class PearsonSSOViewTests(MockedESTestCase, APITestCase):
+    """
+    Tests for the SSO views
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super(PearsonSSOViewTests, cls).setUpTestData()
+        # create an user
+        cls.user = UserFactory()
+
+    def setUp(self):
+        super(PearsonSSOViewTests, self).setUp()
+        self.client.force_login(self.user)
+
+    def test_sso_get_no_exam_profile(self):
+        """
+        Test issuing a GET request when user has no ExamProfile
+        """
+        with patch('exams.views.sso_digest', return_value='test value'):
+            response = self.client.get("/api/v0/pearson/sso/")
+            assert response.status_code == status.HTTP_403_FORBIDDEN
+            assert response.json() == {
+                'error': 'You are not ready to schedule an exam at this time',
+            }
+
+    @ddt.data(
+        (ExamProfile.PROFILE_INVALID, ),
+        (ExamProfile.PROFILE_IN_PROGRESS, ),
+        (ExamProfile.PROFILE_PENDING, ),
+    )
+    @ddt.unpack
+    def test_sso_get_with_exam_profile_not_success(self, profile_status):
+        """
+        Test issuing a GET request when user has an ExamProfile in non-success status
+        """
+        ExamProfile.objects.create(
+            profile=self.user.profile,
+            status=profile_status,
+        )
+
+        with patch('exams.views.sso_digest', return_value='test value'):
+            response = self.client.get("/api/v0/pearson/sso/")
+            assert response.status_code == status.HTTP_403_FORBIDDEN
+            assert response.json() == {
+                'error': 'You are not ready to schedule an exam at this time',
+            }
+
+    def test_sso_get_with_exam_profile_success(self):
+        """
+        Test issuing a GET request when user has an ExamProfile in PROFILE_SUCCESS status
+        """
+        ExamProfile.objects.create(
+            profile=self.user.profile,
+            status=ExamProfile.PROFILE_SUCCESS,
+        )
+
+        with patch('exams.views.sso_digest', return_value='test value'):
+            response = self.client.get("/api/v0/pearson/sso/")
+            result = response.json()
+            assert response.status_code == status.HTTP_200_OK
+
+            timestamp = result['timestamp']
+            assert isinstance(timestamp, int)
```

Comment:
I think it's worth testing...while I doubt we'd intentionally introduce such a regression, I'm not certain on whether or not it would present as an obvious error in production were a regression introduced.

Also, it's probably worth testing purely because if it is wrong the impact is very high - no one would be able to SSO to Pearson to schedule exams.